django/db/migrations/writer.py

```diff
@@ -472,6 +490,8 @@ def serialize(cls, value):
                     "For more information, see "
                     "https://docs.djangoproject.com/en/%s/topics/migrations/#serializing-values"
                     % (value.__name__, module_name, get_docs_version()))
+            if module_name == '__builtin__':
```

Comment:
I guess this is not strictly related to the primary purpose of the commit (as in, the functools case would work without this)?

If so, it would be better to make this a separate pull request for clarity, and it also needs its own test.